src/gpu/align_z_to_other.cu

```diff
@@ -0,0 +1,239 @@
+                    transfer_pixel(depth_pixel_index, y * other_intrin.width + x, z_pixels_gpu, out_pixels_gpu);
+        }
+    }
+
+
+    class CudaStreamWrapper {
+    private:
+        cudaStream_t stream;
+    public:
+        CudaStreamWrapper() {
+            cudaStreamCreateWithFlags(&stream, cudaStreamNonBlocking);
+            DEBUG("GPU Stream Created");
+
+        }
+
+        ~CudaStreamWrapper() {
+            cudaStreamDestroy(stream);
+            DEBUG("GPU Stream Destroyed");
+
+        }
+
+        operator cudaStream_t&(){
+            return stream;
+        }
+
+
+    };
+    static std::mutex gpu_mutex;
+    __host__ bool align_z_to_other(rsimpl::byte * z_aligned_to_other, const uint16_t * z_pixels, float z_scale, const rs_intrinsics & z_intrin, const rs_extrinsics & z_to_other, const rs_intrinsics & other_intrin)
+    {
+        DEBUG("GPU NOT DISABLED");
+
+        int deviceCount = 0;
+        auto cudaCallErrorStatus = cudaGetDeviceCount(&deviceCount);
+        if (cudaSuccess != cudaCallErrorStatus) {
+            std::cerr << "Failed to obtain number of GPUs. " << cudaGetErrorName(cudaCallErrorStatus) << " " << cudaGetErrorString(cudaCallErrorStatus) << std::endl;
+            return false;
+        }
+        if (deviceCount == 0) {
+            DEBUG("NO ENABLED GPUs PRESENT");
+            return false;
+        }
+
+
+        // ignore this error. technically the blocking sync can only be set once prior to device initialization. once
+        // that initial setting is done, subsequent settings result in an error until device is restarted
+        unsigned int cudaFlags = 0;
+        cudaCallErrorStatus = cudaGetDeviceFlags(&cudaFlags);
+        DEBUG("Flags is set to: " << cudaFlags);
+        DEBUG("Error status is: " << cudaCallErrorStatus);
+        if (!(cudaFlags & cudaDeviceScheduleBlockingSync)) {
+            DEBUG("attempting to set the cudaDeviceScheduleBlockingSync flag");
+            std::lock_guard<std::mutex> lock(gpu_mutex);
+            // check flag again to make sure we don't reset it after some other thread set it for us
+            cudaCallErrorStatus = cudaGetDeviceFlags(&cudaFlags);
+            if (!(cudaFlags & cudaDeviceScheduleBlockingSync)) {
+                DEBUG("flag still not set");
+                cudaCallErrorStatus = cudaSetDeviceFlags(cudaDeviceScheduleBlockingSync);
+                if (cudaCallErrorStatus != cudaSuccess) {
+                    DEBUG("reseting device");
+                    cudaDeviceSynchronize();
+                    cudaDeviceReset();
+                    cudaCallErrorStatus = cudaSetDeviceFlags(cudaDeviceScheduleBlockingSync);
+                }
+            }
+        }
+
+
+        auto out_z = (uint16_t *)(z_aligned_to_other);
+
+        auto cudaDeleterUnsignedInt = [](unsigned int *ptr) {cudaFree(ptr);};
+        auto cudaDeleterUInt16t = [](uint16_t *ptr) {cudaFree(ptr);};
+        CudaStreamWrapper myCudaStream;
```

Comment:
FYI, the versions of g++ and clang++ we're using both support c++14. You could replace the two custom deleters with a single lambda with an auto type parameter. (The changes for the firmware bootloader bumped our --std parameter from 11 to 14. I think librealsense has it's own CMakeList.txt rules for this)